src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -261,25 +261,47 @@ private EmbeddingProvidersConfig.EmbeddingProviderConfig getAndValidateProviderC
     return providerConfig;
   }
 
-  // TODO: 1. remove the first if statement when fully support validateAuthentication
-  //  2. Check if user authentication type is support
-  //  3. Check if required token is provided
-  //  4. Check if token is valid
+  /**
+   * Validates user authentication for creating a collection using the specified configurations.
+   *
+   * @param userConfig The vectorize configuration provided by the user.
+   * @param providerConfig The embedding provider configuration.
+   * @throws ApiException If the user authentication is invalid.
+   */
   private void validateAuthentication(
       CreateCollectionCommand.Options.VectorSearchConfig.VectorizeConfig userConfig,
       EmbeddingProvidersConfig.EmbeddingProviderConfig providerConfig) {
+    // Get all the accepted key in auth
+    List<String> acceptedKeys =
+        providerConfig.supportedAuthentications().values().stream()
+            .filter(config -> config.enabled() && config.tokens() != null)
+            .flatMap(config -> config.tokens().stream())
+            .map(EmbeddingProvidersConfig.EmbeddingProviderConfig.TokenConfig::accepted)
+            .toList();
+
+    // If the user hasn't provided authentication details, verify that the 'NONE' authentication
+    // type is enabled.
     if (userConfig.authentication() == null) {
-      return;
+      if (!providerConfig
+          .supportedAuthentications()
```

Comment:
There could be NPE if no entry for `NONE` in YAML config?

Reply:
Not sure if we validate our provider config to make sure every entry always has `NONE`: if not, I think we must assume here that missing definition means same as disabled one.